Add webpacker compatibility with config switch and installation generator (#5855)
activeadmin/activeadmin: docs/0-installation.md

If you're getting the error `wrong number of arguments (6 for 4..5)`, [read #2703].

## webpacker

> Since rails 6.0 webpacker has become the default asset generator. You can opt in using webpacker for ActiveAdmin assets as well by setting `use_webpacker` at install or manually.

Reviewer: I'm not a great prose writer, but I added a small suggestion that maybe reads a bit better?

```suggestion
Since rails 6.0, webpacker has become the default asset generator. You can opt into using webpacker for ActiveAdmin assets as well by setting `use_webpacker` at installation time or manually.
```

> Since rails 6.0, webpacker has become the default asset generator. You can opt into using webpacker for ActiveAdmin assets as well by setting `use_webpacker` at installation time or manually.

Reviewer 2: Just a slight difference since its technically new apps built with Rails 6 that get the default but not if you upgrade. I think this helps be a little clearer as to why its opt in for AA and would help avoid confusion.

```suggestion
For new apps starting with Rails 6.0, Webpacker has become the default asset generator. You can **opt-in to using Webpacker for ActiveAdmin assets** as well by updating your configuration to turn on the `use_webpacker` option, either at installation time or manually.
```

If you're getting the error `wrong number of arguments (6 for 4..5)`, [read #2703].

## webpacker

Since rails 6.0 webpacker has become the default asset generator. You can opt in using webpacker for ActiveAdmin assets as well by setting `use_webpacker` at install or manually.

* at active_admin installation:

  ```sh
  rails g active_admin:install --use_webpacker
  ```

* manually:

  ```ruby
  ActiveAdmin.setup do |config|
    config.use_webpacker = true
  end
  ```

  And run the generator to get default Active Admin assets:

  ```sh
  rails g active_admin:webpacker
  ```

You will need to add `@activeadmin/activeadmin` package to your `package.json`:

```sh
yarn add @activeadmin/activeadmin
```

Either way, Active Admin needs jquery to be available through webpack plugin, so if you haven't already, add this to your `config/webpack/environment.js`:

```javascript
const webpack = require('webpack')
environment.plugins.prepend('Provide',
  new webpack.ProvidePlugin({
    "$":"jquery",
    "jQuery":"jquery",
    "window.jQuery":"jquery"
  })
)

> ```

Reviewer 2: Since we have a webpacker generator now (thanks!) could we not include this there? There is an option to insert into an existing file or append to a file so this should be easy to automate. I'd suggest going off [some of the existing install files in webpacker](https://github.com/rails/webpacker/tree/master/lib/install) to get ideas. With that change then you'd also be able to remove this code from the support template too.

Author: Good point! PR updated accordingly
